### PR TITLE
Sign collector and update visibility

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
contributes to https://github.com/tonerdo/coverlet/issues/555

I've signed collector and updated all visibility of object.
This is a first step to expose coverage.core.dll as a lib, we'll expose only public pertinent object through interfaces/factories to be free to update internal implementation.

AFTER MERGE WE MUST UPDATE VSTEST RUNTIME AND DO SOME TESTS https://github.com/microsoft/vstest/pull/2020/files

cc: @tonerdo @vagisha-nidhi @petli

### CODE REVIEW

**Comment by MarcoRossignoli** [comment]
coverlet.collector.snk keys
```
C:\git\coverlet\src\coverlet.collector (signcollector -> origin)
λ sn -tp token.snk

Microsoft (R) .NET Framework Strong Name Utility  Version 4.0.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Public key (hash algorithm: sha1):
00240000048000009400000006020000002400005253413100040000010001003d23b9ef372215
da7c81af920b919db5799fd021a1ca10b2e9e0ddac71237a29f8f6361a805a747457e561a7d616
417f1870cda099486df25d580a4e11a0738293342881566254d7840e42f42fb9bfd8e8dca354df
7dc68db14b2d0cd79bb2bf7afdbd62bd948d81b534cba7a326cf6ee840a1aee5dba0a1c660b308
13ca99e5

Public key token is 4beb1ad7f05a1c71

C:\git\coverlet\src\coverlet.collector (signcollector -> origin)
λ sn -t token.snk

Microsoft (R) .NET Framework Strong Name Utility  Version 4.0.30319.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Public key token is 4beb1ad7f05a1c71
```

File: src/coverlet.core/Reporters/IReporter.cs [hunk 1]
```diff
@@ -1,14 +1,14 @@
 namespace Coverlet.Core.Reporters
 {
-    public interface IReporter
+    internal interface IReporter
```
Comment (petli) [inline src/coverlet.core/Reporters/IReporter.cs]: I'm not sure in what situations coverlet would be used as a lib, but could one be that the user want to provide their own IReporter?  Or would they just get CoverageResult directly from coverlet and do whatever they need to do with that without going through the reporting interface?

Comment (MarcoRossignoli) [reply src/coverlet.core/Reporters/IReporter.cs]: In past we had a discussion on it https://github.com/tonerdo/coverlet/issues/212 some guys wants integrate the feature as native.

Comment (MarcoRossignoli) [reply src/coverlet.core/Reporters/IReporter.cs]: Do you agree?What do you think?

Comment (petli) [reply src/coverlet.core/Reporters/IReporter.cs]: Ok, so for now hide everything, then in a separate issue start designing externally viable APIs?  Sounds good.

Comment (MarcoRossignoli) [reply src/coverlet.core/Reporters/IReporter.cs]: Yep

**Comment by MarcoRossignoli** [comment]
> I have reservations about exposing coverlet.core publicly

@tonerdo I don't have a strong opinion about it...I don't know if this scenario makes sense.
If you remember we talked about it some time ago https://github.com/tonerdo/coverlet/issues/212 and also in there I wasn't so "happy" about.
We can decide to wait for more community pressure. 
BTW do you agree in hiding all implementation to be more clean?
I like the idea that if someone add reference directly to coverlet libs doesn't see anything.
@petli thought?

**Comment by MarcoRossignoli** [comment]
@vagisha-nidhi merged FYI

**Pull request merged**